sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -270,23 +293,36 @@ private static void killUnneededTasks(StateStore stateStore, Set<String> taskToD
                 .collect(Collectors.toSet());
 
         // Clear the TaskIDs from the TaskInfos so we drop all future TaskStatus Messages
+        //TODO: Investigate why this is needed anymore.
```

Comment:
If this is still needed, should it make its way to the new `killIUnneededTasks` method?

Reply:
At this point my best guess is that this code would be run if, for example, an entire class of pods or tasks was removed from the service spec, as part of e.g. an upgrade. It isn't something that we've been exercising, so it's not even clear if this would work as intended.

I'm sorta on the fence about whether it even makes sense to keep around behind an envvar, but at the same time it doesn't really hurt anything to do so.